sdk/azcore/policy_bearer_token.go

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcore
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	bearerTokenPrefix = "Bearer "
+)
+
+// DO NOT USE. This is for internal SDK usage. The BearerTokenProcessor is used to provide
+// custom implementations for the bearer token policy.
+type BearerTokenProcessor interface {
+	// IsZeroOrExpired returns a bool if there is a tenant that needs to be initialized, it also
+	// returns a list of tenants to get a token for
+	IsZeroOrExpired(map[string]time.Time) (bool, []string)
+	// ShouldRefresh returns a bool is a token needs to be refreshed and the list of tenants that
+	// need to be refreshed
+	ShouldRefresh(map[string]time.Time) (bool, []string)
+	// GetToken performs the GetToken call and returns the tokens that were retreived
+	GetToken(ctx context.Context, cred TokenCredential, tenants []string, opts TokenRequestOptions) (map[string]*AccessToken, error)
+	// Header returns the key for the header in the request
+	Header() string
+}
+
+type bearerTokenPolicy struct {
+	// cond is used to synchronize token refresh.  the locker
+	// must be locked when updating the following shared state.
+	cond *sync.Cond
+
+	// renewing indicates that the token is in the process of being refreshed
+	renewing bool
+
+	// header contains the authorization header value
+	header string
+
+	// expiresOn is when the token will expire
+	expiresOn map[string]time.Time
```

Comment:
I think this, and `header` above should be moved into their specific policies.

Reply:
I think moving expiresOn works well if we move it to the specific policy...but the header is used in the initial conditional on the refresh policy...so I think that one should stay as is. I'll work on removing the expiresOn from the policy.

Reply:
Done